components/Search.tsx

```
@@ -38,40 +84,111 @@ const Search = () => {
-      const files = await getFiles({
+      // Run keyword search always
+      const keywordPromise = getFiles({
         types: [],
         searchText: debouncedQuery,
         limit: 8,
       });
 
+      // Run semantic search only if query looks semantic
+      const shouldRunSemantic = isSemanticQuery(debouncedQuery);
+      const semanticPromise = shouldRunSemantic
+        ? fetchSemanticResults(debouncedQuery, controller.signal)
```

ESLint

[error] Error: Cannot access variable before it is declared

`fetchSemanticResults` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/components/Search.tsx:102:11
  100 |       const shouldRunSemantic = isSemanticQuery(debouncedQuery);
  101 |       const semanticPromise = shouldRunSemantic
> 102 |         ? fetchSemanticResults(debouncedQuery, controller.signal)
      |           ^^^^^^^^^^^^^^^^^^^^ `fetchSemanticResults` accessed before it is declared
  103 |         : Promise.resolve([]);
  104 |
  105 |       let files = { documents: [] as FileItem[] };

/home/jailuser/git/components/Search.tsx:144:3
  142 |   }, [debouncedQuery, path, router, searchParams]);
  143 |
> 144 |   const fetchSemanticResults = async (
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 145 |     q: string,
      | ^^^^^^^^^^^^^^
> 146 |     signal?: AbortSignal,
      …
      | ^^^^^^^^^^^^^^
> 162 |     }
      | ^^^^^^^^^^^^^^
> 163 |   };
      | ^^^^^ `fetchSemanticResults` is declared here
  164 |
  165 |   const handleClickItem = (file: FileItem) => {
  166 |     setOpen(false);

(react-hooks/immutability)